src/main/kotlin/org/eclipse/uprotocol/uri/serializer/MicroUriSerializer.kt

```diff
@@ -59,12 +60,12 @@ class MicroUriSerializer private constructor() : UriSerializer<ByteArray> {
      * @param uri The [UUri] data object.
      * @return Returns a byte[] representing the serialized [UUri].
      */
-    override fun serialize(uri: UUri?): ByteArray {
-        if (uri == null || uri.isEmpty() || !uri.isMicroForm()) {
+    override fun serialize(uri: UUri): ByteArray {
+        if (uri.isEmpty() || !uri.isMicroForm()) {
```

Comment:
Is this condition right? if the uri is not micro form, we serialize with an empty array? It's not related to your patch but that seems odd.

Reply:
it's from MicroUriSerializer, I think it's reasonable to serialize with an empty array if it is not in micro form.